packages/react-components/src/components/xy-output-overview-component.tsx

```diff
@@ -334,4 +342,15 @@ export class XYOutputOverviewComponent extends AbstractXYOutputComponent<Abstrac
     protected getTitleBarTooltip(): string {
         return this.props.outputDescriptor.name + ' overview';
     }
+
+    protected showOptionsMenu(): React.ReactNode {
+        return <React.Fragment>
+            {<div className='options-menu-container'>
+                <button title="Select overview output source" className='options-menu-button'
+                onClick={()=>{signalManager().fireSelectOverviewOutputSignal(this.props.experiment);}}>
```

Comment:
Shouldn't the onClick handler open the dialog and when the use select an XY output a signal is sent with the new output descriptor. You can use the tspClient that is passed to the output to fetch the available outputs. No need to pass the experiment to the class.

Reply:
The reason why I did this was because in the trace-viewer, there is a handler that listens to the SelectOverviewOutputSignal for each opened trace. When we have multiple traces open, multiple dialogs will be opened by this listener. Thus, I use the experiment UUID to distinguish which experiment is opening the selection dialog. I can take a look at the code to see if there is a better option.